R-package/R/utils.R

```diff
@@ -197,30 +197,22 @@ lgb.check.eval <- function(params, eval) {
 lgb.check.wrapper_param <- function(main_param_name, params, alternative_kwarg_value) {
 
   aliases <- .PARAMETER_ALIASES()[[main_param_name]]
-  aliases_provided <- names(params)[names(params) %in% aliases]
-  aliases_provided <- aliases_provided[aliases_provided != main_param_name]
+  aliases_provided <- aliases[aliases %in% names(params)]
 
-  # prefer the main parameter
-  if (!is.null(params[[main_param_name]])) {
-    for (param in aliases_provided) {
-      params[[param]] <- NULL
-    }
-    return(params)
-  }
-
-  # if the main parameter wasn't provided, prefer the first alias
   if (length(aliases_provided) > 0L) {
-    first_param <- aliases_provided[1L]
-    params[[main_param_name]] <- params[[first_param]]
-    for (param in aliases_provided) {
-      params[[param]] <- NULL
+    # set the first alias (has the highest priority) and could be the main param
+    params[[main_param_name]] <- params[[aliases_provided[[1L]]]]
+    # remove the aliases
+    for (alias in aliases_provided) {
+      if (alias != main_param_name) {
+        params[[alias]] <- NULL
+      }
     }
-    return(params)
+  } else {
+    # if not provided in params at all, use the alternative value provided
+    # through a keyword argument from lgb.train(), lgb.cv(), etc.
+    params[[main_param_name]] <- alternative_kwarg_value
   }
-
-  # if not provided in params at all, use the alternative value provided
```

Comment:
What was the reason to switch this from using early `return` statements to an `if-else`? It doesn't seem to me that that change is necessary to accomplish the goals of this PR. If you don't have a strong reason I'd prefer to continue using early returns in this function.